lib/private/Config.php

```
@@ -109,6 +112,7 @@ public function getValue($key, $default = null) {
 	 *
 	 * @param array $configs Associative array with `key => value` pairs
 	 *                       If value is null, the config key will be deleted
+	 * @throws HintException
```

shouldn't this be on the interface? or is this just some internal knowledge and we expect the calling site to fail hard when they run into this?

---

ah, @nickvergessen already asked about this: https://github.com/nextcloud/server/pull/30130#issuecomment-987830088

---

You're right, the interface should have the annotation as well. Added now. Is that, what you meant?

---

Yes.

---

Perfect, thanks :blush: Anything else you'd like to change before approving the PR?